DMFT/src/main/java/com/gov/rsi/DMFT/models/Report.java

```diff
@@ -1,48 +1,47 @@
 package com.gov.rsi.DMFT.models;
 
-import org.bson.types.ObjectId;
+import java.time.LocalDateTime;
+
 import org.springframework.data.annotation.Id;
+import org.springframework.data.mongodb.core.index.Indexed;
 import org.springframework.data.mongodb.core.mapping.Document;
 
+import lombok.Data;
+
+/**
+ * A Report wraps the DMER JSON document along with status and timestamps to 
+ * implement a queue
+ */
 @Document
+@Data
 public class Report {
 	
+	public enum Status {NEW, INPROCESS, FINISHED} 
+	
 	@Id
-	private ObjectId _id;
+	private String			id;
 	
-	private String status;
+	private Status 			status;
+	private LocalDateTime	timeNew;
+	private LocalDateTime	timeInProcess;
+	private LocalDateTime	timeFinished;
 	
-	private Object report;
-
-	public Report(ObjectId _id, String status, Object report) {
-		this._id = _id;
-		this.status = status;
-		this.report = report;
-	}
-
-	public ObjectId get_id() {
-		return _id;
-	}
-
-	public void set_id(ObjectId _id) {
-		this._id = _id;
-	}
-
-	public Object getReport() {
-		return report;
-	}
+	@Indexed(unique = true)
+	private String			licenseNumber;	
+	
+	private String 			json;
 
-	public void setReport(Object report) {
-		this.report = report;
+	public Report(String licenseNumber, String json) {
+		this.licenseNumber = licenseNumber;
+		this.json = json;
 	}
 	
-	public String getStatus() {
-		return status;
-	}
-
-	public void setStatus(String status) {
+	public void setStatus(Status status) {
 		this.status = status;
+		switch (status) {
+			case NEW: 		timeNew = LocalDateTime.now(); break;

```

Comment:
Generally it's best practice to store dates as UTC rather than local.

Reply:
Turns out MongoDB in Spring has an issue storing zoned time. Seems bizarre. Discussed here: https://jira.spring.io/browse/DATAMONGO-2106?focusedCommentId=162512&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-162512
Can be worked around with custom converters, but I'd prefer to put that off for now. Add to backlog for future task?

Reply:
We can put it as Technical Debt for a future iteration.